Source/SwiftLintBuiltInRules/Rules/Style/AttributesRule.swift

```diff
@@ -8,7 +8,8 @@ struct AttributesRule: SwiftSyntaxRule, OptInRule, ConfigurationProviderRule {
         name: "Attributes",
         description: """
             Attributes should be on their own lines in functions and types, but on the same line as variables and \
-            imports
+            imports except when they have properties and attributes_with_arguments_always_on_line_above is true \
+            which is the default
```

Comment:
This seems a bit too long for a violation message. Better to have a dedicated message for the cases where the option comes into play.

Reply:
Does the codebase have an example that I can reference to? I tried to do something like that on my first attempt but I saw the message is a static let and the other rules I saw didn't have something like that. I totally agree it'd be the best option!

Reply:
This list of violations in a rule either accepts a violation with its own message and severity or just a position in which case the rule's description is taken as the message.

An example would be `FunctionParameterCountRule`. In line 76, a `ReasonedRuleViolation` is appended to the list of violations.

Reply:
@SimplyDanny thanks for the help! I just pushed some changes with your proposed approach and added some tests. Let me know if it looks good!